static/app/utils/replays/playback/providers/useCurrentHoverTime.tsx

```diff
@@ -0,0 +1,16 @@
+import type {Dispatch, ReactNode, SetStateAction} from 'react';
+import {createContext, useContext, useState} from 'react';
+
+type ContextType = [undefined | number, Dispatch<SetStateAction<number | undefined>>];
+
+const context = createContext<ContextType>([undefined, () => {}]);
+
+export function ReplayCurrentTimeContextProvider({children}: {children: ReactNode}) {
+  const state = useState<undefined | number>(undefined);
```

Comment:
It's still using state, just like before.
The `currentTime` has the potential to update every 16ms and cause lots of re-renders... but i think for this one, because it's driven by mouse-hover, it'll still cause lots of re-renders as the mouse moves around, but it's directly related to what the user is doing. 
If we want to improve perf in the future we can add a slight debounce to the `setState` call, and/or we could have this accept a callback instead of return the `state` value, the callback would allow us to consume the value and transform it into something more stable/memoizeable depending on the callsite.

For now, i'm happy! `<ReplayContext>` is smaller